flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalogBase.java

```diff
@@ -407,4 +443,201 @@ private Table getHiveTable(ObjectPath tablePath) throws TableNotExistException {
+		throws TableNotExistException, TableNotPartitionedException, PartitionSpecInvalidException, PartitionAlreadyExistsException, CatalogException {
+
+		validateCatalogPartition(partition);
+
+		Table hiveTable = getHiveTable(tablePath);
+
+		ensurePartitionedTable(tablePath, hiveTable);
+
+		try {
+			client.add_partition(createHivePartition(hiveTable, partitionSpec, partition));
+		} catch (AlreadyExistsException e) {
+			if (!ignoreIfExists) {
+				throw new PartitionAlreadyExistsException(catalogName, tablePath, partitionSpec);
+			}
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to create partition %s of table %s", partitionSpec, tablePath));
+		}
+	}
+
+	@Override
+	public void dropPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSpec, boolean ignoreIfNotExists)
+		throws PartitionNotExistException, CatalogException {
+		try {
+			Table hiveTable = getHiveTable(tablePath);
+			ensurePartitionedTable(tablePath, hiveTable);
+			client.dropPartition(tablePath.getDatabaseName(), tablePath.getObjectName(),
+				getFullPartitionValues(tablePath, partitionSpec, getFieldNames(hiveTable.getPartitionKeys())), true);
+		} catch (NoSuchObjectException e) {
+			if (!ignoreIfNotExists) {
+				throw new PartitionNotExistException(catalogName, tablePath, partitionSpec);
+			}
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to drop partition %s of table %s", partitionSpec, tablePath));
+		} catch (TableNotPartitionedException | TableNotExistException | PartitionSpecInvalidException e) {
+			throw new PartitionNotExistException(catalogName, tablePath, partitionSpec, e);
+		}
+	}
+
+	@Override
+	public List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath)
+		throws TableNotExistException, TableNotPartitionedException, CatalogException {
+		Table hiveTable = getHiveTable(tablePath);
+
+		ensurePartitionedTable(tablePath, hiveTable);
+
+		try {
+			return client.listPartitionNames(tablePath.getDatabaseName(), tablePath.getObjectName(), (short) -1).stream()
+				.map(HiveCatalogBase::createPartitionSpec).collect(Collectors.toList());
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to list partitions of table %s", tablePath), e);
+		}
+	}
+
+	@Override
+	public List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
+		throws TableNotExistException, TableNotPartitionedException, CatalogException {
+		Table hiveTable = getHiveTable(tablePath);
+
+		ensurePartitionedTable(tablePath, hiveTable);
+
+		try {
+			// partition spec can be partial
+			List<String> partialVals = MetaStoreUtils.getPvals(hiveTable.getPartitionKeys(), partitionSpec.getPartitionSpec());
+			return client.listPartitionNames(tablePath.getDatabaseName(), tablePath.getObjectName(), partialVals,
```

Comment:
Similar question as above.

Reply:
Same reason for `listPartitions(ObjectPath tablePath)`